components/footer/footer.tsx

```diff
@@ -0,0 +1,73 @@
+import Image from 'next/image';
+
+import style from './footer.module.scss';
+
+export default function Footer() {
+  return (
+    <section id="footer">
+      <div className={style.container}>
+        <div className={style.footer}>
+          <a href="#" className={style.logo}>
+            <Image
+              alt="KedroLogo"
+              src="/images/kedro-logo.svg"
+              width={30}
+              height={30}
+            />
+            <h4 className={style.logoText}>Kedro</h4>
+          </a>
+          {/*TO DO - Update All Links*/}
+          <nav className={style.menu}>
+            <a href="#why-kedro" className={style.link}>
+              Why Kedro?
+            </a>
+            <a href="#features" className={style.link}>
+              Features
+            </a>
+            <a href="#" className={style.link}>
+              Get Started
+            </a>
+            <a
+              href="https://kedro.readthedocs.io/en/stable/"
+              className={style.link}
+              target="_blank"
+              rel="noopener noreferrer"
+            >
+              Documentation
+            </a>
+            <div className={style.iconLinks}>
+              <a
+                className={style.link}
+                href="https://discord.gg/4qeKKspFf8"
+                target="_blank"
+                rel="noopener noreferrer"
+              >
+                <Image
+                  alt="DiscordLogo"
+                  src="/images/discord.svg"
+                  width={30}
+                  height={30}
+                />
+                <span className={style.iconText}>Discord</span>
+              </a>
+              <a
+                className={style.link}
+                href="https://github.com/kedro-org"
+                target="_blank"
+                rel="noopener noreferrer"
+              >
+                <Image
+                  alt="GithubLogo"
+                  src="/images/github.svg"
+                  width={30}
+                  height={30}
+                />
+                <span className={style.iconText}> Github</span>
+              </a>
+            </div>
+          </nav>
+        </div>
```

Comment:
On double checking with the header component, everything here is identical to the main code - please reuse the existing header component rather than having redundant code here.

I notice from the design that the height of the footer is wrapped in a larger container, in which you can import the existing header and wrap that in the `footer` container that you had set up here.

Reply:
It's close, though they aren't exactly identical. There's an important difference [here](https://github.com/kedro-org/kedro-website/blob/main/components/header/header.tsx#L22-L25) that pertains to the opening and closing of the header on mobile.

I don't know if this optimization is worth going through since it's such a small amount of code? We'd have to do it in partial stages, as in having the markup from above and below that element in a shared file, or create another component that took props that would determine if this is the header or footer and act accordingly.

Reply:
yeah, that's what I thought too. Plus the height of the footer is also different from the header and the way it's aligned in the near centre

Reply:
good point, and I completely see where you are coming from.

For me it's less about the amount of code but more about reusability - as we can derive that there is a pattern to repeat the content for both the header and footer, it's better to strip the content out into one place rather than repeating it to prevent the need to maintain the content in 2 places down the line.

@rashidakanchwala this is a very good chance and exercise for you to see how we can strip / refactor code out for reusability ( and a small taste of how we usually reconsider refactoring towards building a design system, which I know is not where we are heading yet but hopefully we as a team can move more towards that mindset)

Some pointers here as to how I would do it:

a) I would add a `isFooter` prop to be passed into the header component to identify the footer usage ( default to false)

b) the `isFooter` prop is used in the `nav` element to determine the style and the `onClick` button - it can be something like this:

```<nav
          className={(!isFooter && menuOpen) ? `${style.menu} ${style.active}` : style.menu}
          onClick={!isFooter ? () => setMenuOpen(false) : null}
        ```

c) as for the burger / close Icon button, only display it when `isFooter` is false

Hope this is clear, feel free to reach out if you hope to pair on this

Reply:
@studioswong Your point about reusability is perfectly valid, but we shouldn't forget about the ["rule of three"](https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)). Eagerly introducing abstractions can cause more issues than solve problems, so I would refrain from making the header and the footer the same (given that we have clear indications that they will differ in the future).

Here's some info about [The Cost of Wrong Abstractions](https://alexkondov.com/the-cost-of-wrong-abstractions/) and what a [Wrong Abstraction](https://sandimetz.com/blog/2016/1/20/the-wrong-abstraction) is.

A good litmus test for checking whether you are onto a good abstraction is if you can easily come up with a name, which can apply for both entities that you are abstracting away. I can't quickly come up with a name, which will perfectly describe both header and footer, without resorting to something inaccurate (like `Navigation`) or with a conjunction (like `HeaderOrFooter`). So to me this is a good indication that this code repetition is either harmless or it will become much more divergent in the future and is only coincidental at the moment.

If we come up with a third place to use the same element, then it's definitely worth refactoring it. But reusing a component once is not worth refactoring it, not only for the time wasted (which is miniscule), but for setting the stage for an increased complexity in the future where more `if this then that` might be added and slowly convert what is essentially a dumb component into a hard to comprehend monster.